src/ophyd_async/core/_derived_signal.py

```diff
@@ -50,8 +50,12 @@ def __init__(
             # Populate received parameters and types
             # Use Signal datatype, or Locatable datatype, or set type as None
             received = {
-                k: v.datatype if isinstance(v, Signal) else get_locatable_type(v)
-                for k, v in raw_and_transform_devices.items()
+                k: v.datatype
+                if isinstance(v, Signal)
+                else type(v)
+                if isinstance(v, int | bool | float | str)
+                else get_locatable_type(v)
+                for k, v in raw_and_transform_devices_and_constants.items()
```

Comment:
Could: If we are having to do the type checking here, we could also partition into `raw_and_transform_devices` and `constants` and pass these down to the `DerivedSignalBackend`...